packages/agent/start.ts

```diff
@@ -213,323 +216,380 @@ const localRun = async (): Promise<void> => {
-            validate: (value: string) => {
-              const trimmed = value.trim();
-              if (!trimmed) return 'Please enter a valid message';
-              if (trimmed.toLowerCase() === 'exit') return true;
-              return true;
-            },
-          },
-        ]);
-
-        if (user.toLowerCase() === 'exit') {
-          console.log(chalk.blue('Exiting interactive mode...'));
-          break;
-        }
+      try {
+        spinner.success({
+          text: chalk.black(
+            `Multi-agent configuration loaded from: ${chalk.cyan(agentPath)}`
+          ),
+        });
+        console.log(chalk.dim('\nLaunching multi-agent environment...\n'));
+        const terminateAgents = await launchMultiAgent(
+          agentPath,
+          modelsConfigPath
+        );
 
-        // Start with a message instead of a spinner to allow log display
-        console.log(chalk.yellow('Processing request...'));
+        console.log(
+          chalk.green('\nAll agents have been launched successfully.')
+        );
 
-        try {
-          // Execute through the supervisor agent which will route appropriately
-          await agentSystem.execute(user);
-
-          // Removing duplicate response formatting and logging since it's now handled
-          // consistently in all mode files (interactive.ts, autonomous.ts, hybrid.ts)
-        } catch (error: any) {
-          console.error(chalk.red('Error processing request'));
-          logger.error('Error during agent execution:', error);
-          console.log(
-            createBox(
-              error.message || 'An unknown error occurred during processing.',
-              { title: 'Error', isError: true }
-            )
-          );
-        }
+        process.on('SIGINT', async () => {
+          console.log('\nGracefully shutting down from SIGINT (Ctrl+C)');
+          await terminateAgents();
+          process.exit(0);
+        });
+        await new Promise(() => {});
+      } catch (error) {
+        spinner.error({ text: 'Failed to initialize multi-agent launcher' });
+        logger.error('Error during multi-agent launch:', error);
+        throw error;
       }
-    } else if (agentMode === AGENT_MODES[AgentMode.AUTONOMOUS]) {
-      console.log(chalk.dim('\nStarting autonomous session...\n'));
-      console.log(
-        chalk.dim(`- Config: ${chalk.bold(path.basename(agentPath))}`)
-      );
-      console.log(chalk.yellow('Running autonomous mode...'));
+    } else {
+      let agentSystem: AgentSystem | null = null;
+      let json_config: AgentConfig = await load_json_config(agentPath);
+      if (!json_config) {
+        throw new Error(`Failed to load agent configuration from ${agentPath}`);
+      }
+      const agentMode = json_config.mode;
+      clearScreen();
+      console.log(logo);
+      const spinner = createSpinner('Initializing Agent System').start();
+      const database: DatabaseCredentials = {
```

Comment:
there is some initialization duplication that could be moved to helpers like initializeDatabase(), or

initializeAgentSystem(
  agentPath: string,
  agentMode: string,
  modelsConfigPath: string, 
  database: DatabaseCredentials
)